core/src/edu/asu/c3simulator/util/Product.java

```diff
@@ -44,7 +47,7 @@ private Image getImage(String imagePath)
 	
 	public Image getProductImage()
 	{
-		return productImage;
+		return getImage(imagePath);
```

Comment:
1: **A lot of screens use the product image as icons and add listeners to them** Ah. That makes sense.

2: **Having new images returned will make it possible to add different listeners to the same product's icon** An alternative is to create a new container each time, and add the image to the container. That should reduce load time for the images, and still allow listeners to function individually. However, now that you've explained the reasoning behind it my initial concerns are lessened.